src/api/src/main/java/engineering/everest/lhotse/api/rest/controllers/OrganizationsController.java

```diff
@@ -71,58 +70,49 @@ public OrganizationsController(DtoConverter dtoConverter,
     @ResponseStatus(OK)
     @ApiOperation("Get information for an organization")
     @AdminOrUserOfTargetOrganization
-    public OrganizationResponse getOrganization(User requestingUser, @PathVariable UUID organizationId) {
+    public OrganizationResponse getOrganization(@ApiIgnore Principal principal, @PathVariable UUID organizationId) {
         return dtoConverter.convert(organizationsReadService.getById(organizationId));
     }
 
     @PostMapping("/register")
     @ResponseStatus(CREATED)
     @ApiOperation("Register a new organization")
     public OrganizationRegistrationResponse registerOrganization(@RequestBody @Valid RegisterOrganizationRequest request) {
-        var organizationId = randomFieldsGenerator.genRandomUUID();
-        var userId = randomFieldsGenerator.genRandomUUID();
-
-        pendingRegistrationsService.registerOrganization(organizationId, userId, request.getOrganizationName(), request.getStreet(),
-                request.getCity(), request.getState(), request.getCountry(), request.getPostalCode(), request.getWebsiteUrl(),
-                request.getContactName(), request.getContactPhoneNumber(), request.getContactEmail(), request.getContactPassword());
-        return new OrganizationRegistrationResponse(organizationId, userId);
-    }
-
-    @GetMapping("/{organizationId}/register/{confirmationCode}")   // Can't easily POST from an email link
-    @ResponseStatus(OK)
-    @ApiOperation("Confirm the email address used to register an organization")
-    public void confirmOrganizationContactEmail(@PathVariable UUID organizationId, @PathVariable UUID confirmationCode) {
-        pendingRegistrationsService.confirmOrganizationRegistrationEmail(organizationId, confirmationCode);
+        var requestingUserId = randomFieldsGenerator.genRandomUUID();
```

Comment:
There is a new gap now that the email confirmation code flow has been deprecated. It's possible to continuously submit organisation registration requests, each of which creates an organisation.

There is also no email address uniqueness validation happening here.

Reply:
Self registration is broken. Users are not created because the application admin organisation creation flow is triggered.

Reply:
I have removed the register organization api endpoint.
Now self registration only works through keycloak page redirection.